From 0f15d52a0332760c34e67451a64bed43da00a7d5 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Fri, 27 Jun 2025 16:34:43 +0200
Subject: [PATCH] Revert "drm/bridge/synopsys: dsi: support of output rotation"

This reverts commit c21e0f6a983c28873b5cb846d1e625ca0616ae02.
This patch is now useless & be hard to push to upstream.

Change-Id: I4ff47d42d75cd8bf68c309e559cbf1a7a41076fa
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/470705
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c | 49 +++++--------------
 1 file changed, 12 insertions(+), 37 deletions(-)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c b/drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c
index 300b4fb7ce65..8502949554f5 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c
@@ -254,7 +254,6 @@ struct dw_mipi_dsi {
 	u32 lanes;
 	u32 format;
 	unsigned long mode_flags;
-	int rotation;
 
 #ifdef CONFIG_DEBUG_FS
 	struct dentry *debugfs;
@@ -364,12 +363,6 @@ static int dw_mipi_dsi_host_attach(struct mipi_dsi_host *host,
 			return ret;
 	}
 
-	/* check if a rotation is required on panel */
-	ret = of_property_read_u32(bridge->of_node, "rotation", &dsi->rotation);
-	if (ret < 0)
-		/* fail to get rotation, set 0 by default */
-		dsi->rotation = 0;
-
 	return 0;
 }
 
@@ -727,16 +720,11 @@ static void dw_mipi_dsi_video_packet_config(struct dw_mipi_dsi *dsi,
 	 * DSI_VNPCR.NPSIZE... especially because this driver supports
 	 * non-burst video modes, see dw_mipi_dsi_video_mode_config()...
 	 */
-	if (dsi->rotation == 90 || dsi->rotation == 270)
-		dsi_write(dsi, DSI_VID_PKT_SIZE,
-			  dw_mipi_is_dual_mode(dsi) ?
-			  VID_PKT_SIZE(mode->vdisplay / 2) :
-			  VID_PKT_SIZE(mode->vdisplay));
-	else
-		dsi_write(dsi, DSI_VID_PKT_SIZE,
-			  dw_mipi_is_dual_mode(dsi) ?
-			  VID_PKT_SIZE(mode->hdisplay / 2) :
-			  VID_PKT_SIZE(mode->hdisplay));
+
+	dsi_write(dsi, DSI_VID_PKT_SIZE,
+		       dw_mipi_is_dual_mode(dsi) ?
+				VID_PKT_SIZE(mode->hdisplay / 2) :
+				VID_PKT_SIZE(mode->hdisplay));
 }
 
 static void dw_mipi_dsi_command_mode_config(struct dw_mipi_dsi *dsi)
@@ -803,15 +791,9 @@ static void dw_mipi_dsi_line_timer_config(struct dw_mipi_dsi *dsi,
 {
 	u32 htotal, hsa, hbp, lbcc;
 
-	if (dsi->rotation == 90 || dsi->rotation == 270) {
-		htotal = mode->vtotal;
-		hsa = mode->vsync_end - mode->vsync_start;
-		hbp = mode->vtotal - mode->vsync_end;
-	} else {
-		htotal = mode->htotal;
-		hsa = mode->hsync_end - mode->hsync_start;
-		hbp = mode->htotal - mode->hsync_end;
-	}
+	htotal = mode->htotal;
+	hsa = mode->hsync_end - mode->hsync_start;
+	hbp = mode->htotal - mode->hsync_end;
 
 	/*
 	 * TODO dw drv improvements
@@ -832,17 +814,10 @@ static void dw_mipi_dsi_vertical_timing_config(struct dw_mipi_dsi *dsi,
 {
 	u32 vactive, vsa, vfp, vbp;
 
-	if (dsi->rotation == 90 || dsi->rotation == 270) {
-		vactive = mode->hdisplay;
-		vsa = mode->hsync_end - mode->hsync_start;
-		vfp = mode->hsync_start - mode->hdisplay;
-		vbp = mode->htotal - mode->hsync_end;
-	} else {
-		vactive = mode->vdisplay;
-		vsa = mode->vsync_end - mode->vsync_start;
-		vfp = mode->vsync_start - mode->vdisplay;
-		vbp = mode->vtotal - mode->vsync_end;
-	}
+	vactive = mode->vdisplay;
+	vsa = mode->vsync_end - mode->vsync_start;
+	vfp = mode->vsync_start - mode->vdisplay;
+	vbp = mode->vtotal - mode->vsync_end;
 
 	dsi_write(dsi, DSI_VID_VACTIVE_LINES, vactive);
 	dsi_write(dsi, DSI_VID_VSA_LINES, vsa);
-- 
2.53.0

